### ISSUE TITLE
feat(model): preserve fixed-depth KV lineage through fork and rollback

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#11741]
## Current state
Parent #11700 and docs/research/astra-recurrent-depth-support.md identify recurrent-step lineage as a requirement. #11737 deliberately rejects unsupported speculative branch/rollback; it does not implement support. Existing internal/model/pagedkv.go Fork and paged_prefix_cow.go session forks are the working spine, not proof of recurrent correctness.

## Why now
Tier 2, Core. For native recurrent serving users, branching must preserve execution-depth identity rather than treating all token prefixes as interchangeable. Better than disabling speculation only when deterministic parity is proven.

## In scope
Implement one fixed-depth recurrent paged-KV fork and rejected-suffix rollback path that preserves parent lineage and execution-slot identity. Compare the next decode after rollback with a clean non-speculative run. Restrict to the fixed two-loop Nanbeige envelope established by #11728.

## Out of scope
Adaptive Huginn branching, speculative proposal policy, heterogeneous batching, performance claims, allocator redesign, and proprietary Astra architecture claims.

## Likely files
- `internal/model/pagedkv.go`
- `internal/model/paged_prefix_cow.go`
- `internal/model/recurrent_branch_integration_test.go`

## Dependencies
#11729 and #11737. If a separate executor contract is necessary, split and link that prerequisite before expanding this write set.

## Working spine
Existing PagedKV.Fork and PagedPrefixSession.Fork; retain fail-closed admission from #11737 until this envelope has a witness.

## Acceptance gate
One proposed real-path witness: `go test ./internal/model -run '^TestRecurrentBranchRollbackNativeParity$' -count=1`. Execute on sanctioned WSL/fleet with the actual pinned Nanbeige BF16 checkpoint from the research note. Fork at a fixed prefix, execute and reject a suffix, resume, and compare logits with clean recomputation at a declared tolerance. Assert parent state unchanged and correct per-depth slot lineage. Missing checkpoint or skipped execution is not success.

## Closure binding
Close only with implementation commit and captured green witness naming checkpoint revision, configuration, engine=fak-native, tolerance and measured result; reference that evidence on #11700. No implementation or parity is claimed by this ticket.

## Operating envelope and rollback
22 physical layers, two loops, unshared execution-slot KV; one parent and one child branch. Preserve rejection outside the supported envelope. Do not restart servers with active dependent sessions.

Expected steps: 3
Work unit: leaf

## Parent context
#11700

## Done condition
The single in-scope outcome passes the acceptance gate with execution evidence attached; no mock or skipped runtime satisfies closure.

## Witness
`go test ./internal/model -run '^TestRecurrentBranchRollbackNativeParity$' -count=1`
